### ISSUE TITLE
[TECHDEBT] Enforce code formatting style

### ISSUE THREAD

**Issue by kalemas** [bibletime/bibletime#364]
# Summary
When reading existing code it is hard to tell if how exactly should be formatted code. Following wiki [page](http://bibletime.info/historic-devel.bibletime.info/wiki/Programming_Guidelines.html) did not cover all possible cases and still leave responsibility on user. I think setting up automatic formatter would be quite useful, result it higher quality code.

# Remediation approach
I ve just starter to use VS Code for Cpp development and found that `clang-format` is default enabled auto formatter, i think, providing suitable configuration would easily resolve this issue. Here is [configuration](https://github.com/kalemas/mini/blob/5fffb580df34b3e8d81333dba0749fe38aa51906/.clang-format#) i started when prepared recent PRs.

# Notes
I also found very useful `pre-commit` tool, i think there would be something useful for cpp development, like [here] (https://github.com/pocc/pre-commit-hooks#example-usage).

**Comment by jaakristioja** [comment]
Agreed.

But the problem with tools like `clang-format` is that they never work the way I want. They are simply too limited. I just tried again with a custom `.clang-format`. The results were quite terrible and some were not fixable with changes to the `.clang-format`. There are also many exceptions one simply can not configure properly, e.g. formatting of Qt's `connect` or BibleTime's `BT_CONNECT` statements, which I think often look best as something like:

```c++
BT_CONNECT(signalSender, &Sender::someSignal,
           target,       &target::someSlotWithSlightlyLongerName);
```

I will however consider writing a new style guide.

PS: Don't rely on the historic wiki. It is named "historic" for a reason :)

**Comment by kalemas** [comment]
Yes. But i d still rely on automatic formatter for 95% of cases and do manual fixes as needed. It follows from your message that there is no actual code formatting style guide, so it is better to not propose any clang-format configuration now.
